gef-ui/server/controllers/unissued-facilities/index.js

```diff
@@ -0,0 +1,516 @@
+  } catch (err) {
+    return res.render('partials/problem-with-service.njk');
+  }
+};
+
+// when changing unissued facility from application preview page
+const changeUnissuedAboutFacilityChange = async (req, res) => {
+  const { params, query } = req;
+
+  try {
+    const {
+      facilityType,
+      facilityName,
+      hasBeenIssued,
+      monthsOfCover,
+      shouldCoverStartOnSubmission,
+      issueDateDay,
+      issueDateMonth,
+      issueDateYear,
+      coverStartDateDay,
+      coverStartDateMonth,
+      coverStartDateYear,
+      coverEndDateDay,
+      coverEndDateMonth,
+      coverEndDateYear,
+      facilityTypeString,
+      dealId,
+      facilityId,
+      status,
+      change,
+    } = await renderChangeFacilityPartial(params, query, true);
+    return res.render('partials/unissued-change-about-facility.njk', {
+      facilityType,
+      facilityName,
+      hasBeenIssued,
+      monthsOfCover,
+      shouldCoverStartOnSubmission,
+      issueDateDay,
+      issueDateMonth,
+      issueDateYear,
+      coverStartDateDay,
+      coverStartDateMonth,
+      coverStartDateYear,
+      coverEndDateDay,
+      coverEndDateMonth,
+      coverEndDateYear,
+      facilityTypeString,
+      dealId,
+      facilityId,
+      status,
+      change,
+    });
+  } catch (err) {
+    return res.render('partials/problem-with-service.njk');
+  }
+};
+
+/**
+ * validation for changing facilities
+ * returns required parameters for post update facility
+ */
+const facilityValidation = async (req, res) => {
+  const { body, query, params } = req;
+  const { facilityType } = body;
+  const facilityTypeString = facilityType.toLowerCase();
+  const { saveAndReturn, status } = query;
+  const { dealId, facilityId } = params;
+  const application = await api.getApplication(dealId);
+  const aboutFacilityErrors = [];
+  const issueDateDay = body['issue-date-day'];
+  const issueDateMonth = body['issue-date-month'];
+  const issueDateYear = body['issue-date-year'];
+  const issueDateIsFullyComplete = issueDateDay && issueDateMonth && issueDateYear;
+  const issueDateIsPartiallyComplete = !issueDateIsFullyComplete && (issueDateDay || issueDateMonth || issueDateYear);
+  const issueDateIsBlank = !issueDateDay && !issueDateMonth && !issueDateYear;
+  const coverStartDateDay = body['cover-start-date-day'];
+  const coverStartDateMonth = body['cover-start-date-month'];
+  const coverStartDateYear = body['cover-start-date-year'];
+  const coverStartDateIsFullyComplete = coverStartDateDay && coverStartDateMonth && coverStartDateYear;
+  const coverStartDateIsPartiallyComplete = !coverStartDateIsFullyComplete && (coverStartDateDay || coverStartDateMonth || coverStartDateYear);
+  const coverStartDateIsBlank = !coverStartDateDay && !coverStartDateMonth && !coverStartDateYear;
+  const coverEndDateDay = body['cover-end-date-day'];
+  const coverEndDateMonth = body['cover-end-date-month'];
+  const coverEndDateYear = body['cover-end-date-year'];
```

Comment:
For the req.body destructuring it could be simpler/less constant declerations with e.g:

```js
const {
  'cover-end-date-day': coverEndDateDay,
  'cover-end-date-month': coverEndDateMonth,
} = body;
```

Reply:
done